src/append_types.rs

```diff
@@ -0,0 +1,234 @@
+// Copyright 2015 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under (1) the MaidSafe.net Commercial License,
+// version 1.0 or later, or (2) The General Public License (GPL), version 3, depending on which
+// licence you accepted on initial access to the Software (the "Licences").
+//
+// By contributing code to the SAFE Network Software, or to this project generally, you agree to be
+// bound by the terms of the MaidSafe Contributor Agreement, version 1.1.  This, along with the
+// Licenses can be found in the root directory of this project at LICENSE, COPYING and CONTRIBUTOR.
+//
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.
+//
+// Please review the Licences for the specific language governing permissions and limitations
+// relating to use of the SAFE Network Software.
+
+use maidsafe_utilities::serialisation::serialise;
+use rust_sodium::crypto::sign::{self, PublicKey, SecretKey, Signature};
+use std::collections::BTreeSet;
+use xor_name::XorName;
+use data::{DataIdentifier, verify_detached};
+use error::RoutingError;
+use priv_appendable_data::PrivAppendedData;
+
+/// The type of access filter for appendable data.
+#[derive(Hash, Eq, PartialEq, PartialOrd, Ord, Clone, RustcDecodable, RustcEncodable)]
+pub enum Filter {
+    /// Everyone except the listed keys are allowed to append data.
+    BlackList(BTreeSet<PublicKey>),
```

Comment:
An application would be banning an account from a forum or forum thread.
(Of course someone could just create a new key pair, but it was decided that this might still be desirable.)